moneta-core/src/main/java/org/javamoney/moneta/internal/OSGIServiceProvider.java

```diff
@@ -97,7 +98,7 @@ public int getPriority() {
             LOG.log(Level.SEVERE, "could not create services of type: " + serviceType, e);
         }
         try{
-            for(T service:ServiceLoader.load(serviceType)){
+            for(T service:ServiceLoader.load(serviceType, Monetary.class.getClassLoader())){
```

Comment:
I am not sure if the Java Money implementation will always come from the same class loader as the API. Both classes should be resolvable from the context classloader. And yes, that may cause issues with parallel operations that are executed inside fork join pools. But won't this fix cause issues with deployments where the implementation lives in a separate module? Or inside the application itself?